src/client/Microsoft.Identity.Client/ApiConfig/Executors/ConfidentialClientExecutor.cs

```diff
@@ -71,14 +75,79 @@ public async Task<AuthenticationResult> ExecuteAsync(
             return await handler.RunAsync(cancellationToken).ConfigureAwait(false);
         }
 
+        private void ValidateAndConfigureMtlsPopForAcquireTokenForClient(
+            AcquireTokenForClientParameters clientParameters, 
+            AcquireTokenCommonParameters commonParameters, 
+            RequestContext requestContext)
+        {
+            if (!clientParameters.UseMtlsPop)
+                return;
+
+            ValidateClaimsOrAssertionsUse();
+            ValidateAuthorityType();
+            ValidateCertificatePresence();
+            ValidateRegionPresence(requestContext);
+
+            commonParameters.MtlsCertificate = _confidentialClientApplication.Certificate;
+            commonParameters.AuthenticationOperation = new MtlsPopAuthenticationOperation(_confidentialClientApplication.Certificate);
+            requestContext.ServiceBundle.Config.IsInstanceDiscoveryEnabled = false;
```

Comment:
I disagree on this. We don't need to disable authority validation. You can always validate the non-mtls authority. 

Regional already does all the heavy lifting. Why don't you just rely on that logic?